README.md

```diff
@@ -1,34 +1,117 @@
+[![Checks](https://github.com/ioki-mobility/go-outline/actions/workflows/checks.yml/badge.svg)](https://github.com/ioki-mobility/go-outline/actions/workflows/checks.yml)
+[![MIT](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/ioki-mobility/go-outline/blob/main/LICENSE)
 
-[def]: https://www.getoutline.com/
+# go-outline
 
+Go client and cli for [outline](https://www.getoutline.com/).
 
 # Usage
 
 ## Client
-```golang
-// Create top level client.
-cl := outline.New("https://baseurl/api",&http.Client{},"api key")
 
+### Create a client
 
-// Fetch information about a collection:
-col,err := cl.Collections().Get("collection id").Do(context.Background())
+```go
+cl := outline.New("https://base.url/api", &http.Client{}, "api key")
+```
+
+> **Note**: You can create a new API key in your outline **account settings**.
+
+### Get a collection
+
+```go
+col, err := cl.Collections().Get("collection id").Do(context.Background())
 if err != nil {
 	panic(err)
 }
 fmt.Println(col)
+```
 
 
-// Fetch information about all collections.
-err := cl.Collections().List().Do(context.Background(), func(c *outline.Collection, err error) (bool, error) {
-	fmt.Println(c)
+### Get all collections
+
+```go
+err := cl.Collections().List().Do(context.Background(), func(col *outline.Collection, err error) (bool, error) {
+	fmt.Println(col)
 	return true, nil
 })
 if err != nil {
 	panic(err)
 }
 ```
 
+### Create a collection
+
+```go
+col, err := cl.Collections().Create("collection name").Do(context.Background()) 
+if err != nil {
+	panic(err)
+}
+fmt.Println(col)
+```
+
+There are also **optional** functions for the `CollectionsCreateClient` available:
+```go
+colCreateClient := cl.Collections().Create("collection name")
+colCreateClient.
+	Description("desc"). 
+	PermissionRead(). // or PermissionReadWrite()
+	Color("#c0c0c0").
+	Private(true).
+	Do(context.Background())
+```
+
+### Document Create
+
+```go
+doc := cl.Documents().Create("Document name", "collection id").Do(context.Background())
+```
+
+There re also **optional** functions for the `DocumentsCreateClient` available:
+```go
+docCreateClient := cl.Documents().Create("Document name")
+docCreateClient.
+	Publish(true). 
+	Text("text").
+	ParentDocumentID(DocumentId("parent document id")).
+	TemplateID(TemplateId("templateId")).
+	Template(false).
+	Do(context.Background())
+```
 
 ## CLI
-TBA
\ No newline at end of file
+
+### Build the CLI
+
+```
+go build -o outline ./cli
+```
+
+### Required flags
+
+Any command requires the flags `baseUrl` and `apiKey`.
+You can simply add them with `--baseUrl https://base.Url/api` 
+and `--apiKey sup3rS3cre7Ap1K3Y`.
+
+> **Note**: The `baseUrl` is the URL to your outline instance, followed by `/api`.
+
+### Collections
+
+The basic command to work with collections is:
+```
+outline collections
+```
+
+#### Collections fetch
+
+To fetch a collection and display it in a json string use:
+```
+outline collections fetch [COLLECTION_ID] [flags]
```

Comment:
@StefMa you do not have to add exhaustive cli docs like this...let the cli help docs take care of that. Of course the sections above are okay.

Reply:
What do you mean by that? Honestly, we need some kind of docs for the cli as well. I guess not many will blindly download "a cli" and first discovery via the help command what it can do.
We need some kind of documentation for that upfront. I would even go a step further and would make use of github pages to document it (but this is our of scope for now).

Reply:
I mean giving an example or two plus summarizing overall cli usage is okay. This IMO is enough to give user an idea what it can be used for. But duplicating docs for all possible options and arguments can become a maintenance headache very quickly.

An alternative solution would be to add output of `--help` so that whatever current docs are  there can be seen before downloading. This makes docs under cli package the only source of truth that we would need to maintain. But again this is not a very critical point. We can discuss this separately later.

Reply:
We might want o find out how gh is doing that https://cli.github.com/manual/gh 😁